### ISSUE TITLE
runc --systemd-cgroup run: need to check that StartTransientUnit succeeds

### ISSUE THREAD

**Issue by kolyshkin** [opencontainers/runc#2313]
**TL;DR: runc should check that system was able to create a unit.**

I was working on runc on a Fedora 31 Vagrant box with older selinux packages (`container-selinux-2.117.0-1.gitbfde70a.fc31.noarch`). Command `runc --systemd run xoo` succeeded, but the container process was running in the same cgroup as my shell, and the designated cgroup was empty (see https://github.com/opencontainers/runc/issues/2310 for details).

```console
# runc --systemd run xoo
# runc list
ID          PID         STATUS      BUNDLE         CREATED                          OWNER
xoo         16201       running     /vagrant/tst   2020-04-14T02:23:16.049427501Z   root
# cat /proc/16201/cgroup 
0::/user.slice/user-1000.slice/session-3.scope
# cat /proc/self/cgroup 
0::/user.slice/user-1000.slice/session-3.scope
```

journalctl shows:
```
Apr 14 02:23:16 localhost.localdomain audit[1]: AVC avc:  denied  { setsched } for  pid=1 comm="systemd" scontext=system_u:system_r:init_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=process permissive=0
Apr 14 02:23:16 localhost.localdomain systemd[1]: runc-xoo.scope: Failed to add PIDs to scope's control group: Permission denied
Apr 14 02:23:16 localhost.localdomain systemd[1]: runc-xoo.scope: Failed with result 'resources'.
Apr 14 02:23:16 localhost.localdomain systemd[1]: Failed to start libcontainer container xoo.
```

So, systemd failed to create a unit, but runc did not show any errors (it's the same with podman+crun).

When trying to start a unit with `systemd.run`, it shows an error:
```
# systemd-run --scope sleep 999
Job failed. See "journalctl -xe" for details.
# journalctl -xe
...
Apr 14 02:36:49 localhost.localdomain audit[1]: AVC avc:  denied  { setsched } for  pid=1 comm="systemd" scontext=system_u:system_r:init_t:s0 tcontext=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023 tclass=process permissive=0
Apr 14 02:36:49 localhost.localdomain systemd[1]: foobar.scope: Failed to add PIDs to scope's control group: Permission denied
Apr 14 02:36:49 localhost.localdomain systemd[1]: foobar.scope: Failed with result 'resources'.
Apr 14 02:36:49 localhost.localdomain systemd[1]: Failed to start /usr/bin/sleep 999.
...
```

**Comment by kolyshkin** [comment]
The issue is probably true for systemd v1 aka LegacyManager.

**Comment by kolyshkin** [comment]
Might be related (need to check): https://github.com/opencontainers/runc/issues/1780
